mysql-test/main/alter_table.test

```diff
@@ -2040,3 +2040,21 @@ analyze table t1;
 set @@use_stat_tables= @save_use_stat_tables;
 set @@optimizer_use_condition_selectivity= @save_optimizer_use_condition_selectivity;
 drop table t1;
+
+--echo #
+--echo # MDEV-18154 Deadlock and assertion upon no-op ALTER under LOCK TABLES
+--echo #
+create or replace table t1 (pk int, i int, primary key (pk)) engine myisam;
+create or replace view v1 as select * from t1;
+lock table v1 read, t1 write;
+--error ER_BAD_FIELD_ERROR
+alter table t1 change f1 f2 int;
+set max_statement_time= 1;
+alter table t1 add column if not exists i int after pk;
+set max_statement_time= 0;
+drop table t1;
```

Comment:
Do we need this case `ALTER TABLE t1 ADD PERIOD FOR SYSTEM_TIME(s,e);` like on jira ?

Reply:
The case was minimized. Did you try to reproduce with it?